src/server/dom.ts

```diff
@@ -135,7 +105,7 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
 
   private async _evaluateInMainAndWaitForSignals<R, Arg>(pageFunction: js.Func1<[js.JSHandle<InjectedScript>, ElementHandle<T>, Arg], R>, arg: Arg): Promise<R> {
     const main = await this._context.frame._mainContext();
-    return main.evaluateAndWaitForSignals(pageFunction, [await main.injectedScript(), this, arg]);
+    return main.eval(pageFunction, {arg: [await main.injectedScript(), this, arg], waitForSignals: true});
```

Comment:
This `arg: ` field containing an array of arguments looks odd now that we have `args:` option, consider using that one instead.